gesasso/frontend/src/components/RequestForm/RequestContext/modules/WoollyContext.tsx

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) Cesar Richard 2023.
+ */
+
+import React, { useEffect, useState } from 'react';
+import {
+  Button, Dropdown, Form, Input,
+} from 'semantic-ui-react';
+// eslint-disable-next-line import/no-cycle
+import { RequestContextComponentProps } from '@gesasso/components/RequestForm/RequestContext/modules/RequestContext';
+// eslint-disable-next-line import/no-unresolved
+import { ContextWoolly, ContextWoollyType } from '@gesasso/components/RequestForm/RequestContext/Types';
+import { DateRangePicker } from 'rsuite';
+import { WoollyArticleFormField } from '@gesasso/components/RequestForm/RequestContext';
+
+export interface WoollyContextProps extends RequestContextComponentProps {
+  context: ContextWoolly;
+}
+
+const WoollyContext = ({ context, onContextChange }: WoollyContextProps): React.ReactElement<any> => {
+  const [name, setName] = useState(context?.name);
+  const [description, setDescription] = useState(context?.description);
+  const [location, setLocation] = useState(context?.location);
+  const [type, setType] = useState(context?.type);
+  const [saleStartDate, setSaleStartDate] = useState(context?.saleStartDate);
+  const [saleEndDate, setSaleEndDate] = useState(context?.saleEndDate);
+
+  useEffect(() => {
+    onContextChange({
+      articles: [],
+      location,
+      name,
+      saleEndDate,
+      saleStartDate,
+      type: undefined,
+      description,
+    });
+  }, [description, location, saleEndDate, saleStartDate]);
+
+  const [subsets, setSubsets] = useState([]);
+  const [articleCounter, setArticleCounter] = useState(0);
+
+  const handleChange = (e) => {
+    const newSubsets = [...subsets];
+    newSubsets.map((subset) => {
+      if (subset.key === e.key) {
+        return e;
+      }
+      return subset;
+    });
+    setSubsets(newSubsets);
+  };
```

Comment:
Sinon le retour du map est jamais utilisé
```suggestion
  const handleChange = (e) => {
    const newSubsets = [...subsets].map((subset) => {
      if (subset.key === e.key) {
        return e;
      }
      return subset;
    });
    setSubsets(newSubsets);
  };
```